### ISSUE TITLE
Handle orders with negative amounts

### ISSUE THREAD

**Issue by bilthon** [MostroP2P/mostro#184]
Orders with negative fiat and sats amounts are allowed to be created, and in the case of the negative sats amount a panic is generated when a hodl invoice with negative amount is requested.

Here's a log dump for that event:
```
2024-01-07T17:02:33.766035Z  INFO mostrod::util: DM content: "{\"Order\":{\"version\":1,\"id\":\"d43d4753-618e-416a-b9fc-9ddd57c97c9d\",\"pubkey\":null,\"action\":\"NewOrder\",\"content\":{\"Order\":{\"id\":\"d43d4753-618e-416a-b9fc-9ddd57c97c9d\",\"kind\":\"Sell\",\"status\":\"Pending\",\"amount\":-2000,\"fiat_code\":\"USD\",\"fiat_amount\":5,\"payment_method\":\"Cash\",\"premium\":0,\"created_at\":1704646953}}}}"
2024-01-07T17:02:33.771251Z  INFO mostrod::util: Sending event: Event {
    id: EventId(
        0xf5bc660d3dcd7758418346bdf950682b0e280f6a8650ae4254e25f00d4c5192c,
    ),
    pubkey: XOnlyPublicKey(
        d9725eb47dc4441cec52c21b3fa63f630dbdf6458dc3bd091c75b5faf3fecb2e28bf42c5978329f2d144e3e4cedd8d25e94b9fa8ded22a708c4219cb5336e099,
    ),
    created_at: Timestamp(
        1704646953,
    ),
    kind: EncryptedDirectMessage,
    tags: [
        PubKey(
            XOnlyPublicKey(
                128642fc8945af582de87e69c7d763626a89435cda6ee38ff8e06a0cf21f39498e20aa1b9dd72f537fcb4f4eb0431eeb0bca780cc9934575e7e2d4fb9b77bf8a,
            ),
            None,
        ),
    ],
    content: "hj/s1pICaVwpRsW3rux8wlSSw2G1JI6iiMQFxCFzdk5EVwL/lhNJJyFnO5IB8anD77TvpDi4FFmEMuhkz6X1/iQ8Xjiq1heAVIYMpDf3QAKRGm7L9++oaK8eLY902I9RGNULfyO5H6r1LTm8V4vvnWcLUb6OtfP7yu+tsm1pFmKAZVl5R9IJO2tvs29oXmtl9EFiwuR89K0iMv1/Wa8WSVb+A92M2aYHDXS7Ro7H91hEdQ6CQlQmWecd26s0lQn9nHCeEYC0bRxT9MhfQPZk57niM1EcGk9axdcCU0C5/ULREfWvBjq9HbWhowJDqLkoXrK6hTYuipqY3Nmpql/6P+Gq36+ulQcVq2xHN5bI6p6+YY54rkctkv0lbgTl92dn1K5JCgpIRuQHSXevRm32T0Ngo+yJyyyNKV1Zst3Keqs=?iv=vlrORIjU6t6JD0k/GORwDA==",
    sig: Signature(bb6fa6714b2a64cdeacaa97cf00714925f637de4b7b3bd16fc4cbabbefb7840527e369ec38927d26933a6072a0fef1005fd61562f50b9ba81cedb67c6728d434),
}
thread 'main' panicked at src/lightning/mod.rs:77:14:
Failed to add hold invoice: Status { code: Unknown, message: "payments of negative value are not allowed, value is -2000000", metadata: MetadataMap { headers: {"content-type": "application/grpc"} }, source: None }
```

I'm not sure what the expected behavior is here, but I'd assume that such order creation requests should be rejected with an error.

**Comment by arkanoider** [comment]
Hi @bilthon ,

I think this could be solved with #183 , let's wait that also @grunch review the fix.
Basically now when Mostro create new order checks also if fixed amount ( and also for safety the API quote ) is negative, in that case it raises an error.

**Comment by grunch** [comment]
closing it via https://github.com/MostroP2P/mostro/pull/183
